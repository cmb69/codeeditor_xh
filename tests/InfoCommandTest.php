<?php

namespace Codeeditor;

use ApprovalTests\Approvals;
use PHPUnit\Framework\TestCase;
use Plib\FakeSystemChecker;
use Plib\View;

class InfoCommandTest extends TestCase
{
    public function testRendersPluginInfo(): void
    {
        $sut = new InfoCommand(
            "./plugins/codeeditor/",
            new FakeSystemChecker(true),
            new View("./views/", XH_includeVar("./languages/en.php", "plugin_tx")["codeeditor"])
        );
        $response = $sut();
        $this->assertSame("Codeeditor 2.0", $response->title());
        Approvals::verifyHtml($response->output());
    }

    public function testRendersFailingSystemChecks(): void
    {
        $sut = new InfoCommand(
            "./plugins/codeeditor/",
            new FakeSystemChecker(),
            new View("./views/", XH_includeVar("./languages/en.php", "plugin_tx")["codeeditor"])
        );
        Approvals::verifyHtml($sut()->output());
    }
}
