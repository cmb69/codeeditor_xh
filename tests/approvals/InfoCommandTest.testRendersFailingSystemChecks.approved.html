
<h1>Codeeditor 2.0</h1>
<h2>System check</h2>
<p class="xh_fail">PHP version ≥ 7.1.0</p>
<p class="xh_fail">CMSimple_XH version ≥ 1.7.0</p>
<p class="xh_fail">Plib_XH version ≥ 1.3</p>
<p class="xh_warning">Folder './plugins/codeeditor/config/' writable</p>
<p class="xh_warning">Folder './plugins/codeeditor/css/' writable</p>
<p class="xh_warning">Folder './plugins/codeeditor/languages/' writable</p>
